src/core/src/core_logic/PatchInstaller.py

```diff
@@ -280,7 +283,7 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         attempted_parent_package_install_count_in_batch_patching = self.attempted_parent_package_install_count
         successful_parent_package_install_count_in_batch_patching = self.successful_parent_package_install_count
 
-        if len(packages) == 0:
+        if len(packages) == 0 or self.is_install_patches_in_single_batch_enabled():
```

Comment:
The additional check should not be needed since 'packages' here represents remaining_packages_pending_to_be_installed. So len(packages) will be 0 for a single batch at this point.

Reply:
In this case if any package is not installed in one go, we are not retrying. This condition is to stop execution after one attempt explicitly even if there are pending updates